### PR TITLE
Add style coding guidelines

### PR DESCRIPTION
**Pull request by pauby**
* Updated the style and coding standards
* Fixed markdown issues
* Updated TOC
* Tidied up

### CODE REVIEW

**Comment by HolisticDeveloper** [comment]
Have you considered referencing a third-party style guide, such as the one from [PoshCode](https://poshcode.gitbooks.io/powershell-practice-and-style/)? One difference I see between them is PascalCasing vs camelCasing for function parameters.

**Comment by pauby** [comment]
That is something I did think about and flip flopped about pointing to it. I ultimately decided I didn't want to in this first run because that unofficial style, while fantastic (and I have used it a lot myself) is a live guide and I had concerns that something we want to encourage, and is currently also encouraged in the style guide is then changed later on.

But I think there is mileage on encouraging people to use it as a general guide for PowerShell so I will add it in.

Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -49,21 +44,21 @@ What is generally considered trivial:
...
+    * If approved through the mailing list or in [Gitter chat](https://gitter.im/chocolatey/boxstarter), ensure the accompanying GitHub issue is created with information and a link back to the mailing list discussion (or the Gitter conversation).
+* Once you get a nod from one of the [Chocolatey Team](https://github.com/chocolatey?tab=members), you can start on the feature.
+* Alternatively, if a feature is on the issues list with the [Up For Grabs](https://github.com/chocolatey/boxstarter/issues?q=is%3Aopen+is%3Aissue+label%3A%22Up+For+Grabs%22) label, it is open for a community member (contributor) to patch. You should comment that you are signing up for it on the issue so someone else doesn't also sign up for the work.
 
 ### Set Up Your Environment
- * Visual Studio 2010+ is recommended for code contributions.
- * For git specific information:
+
+* Visual Studio 2010+ is recommended for code contributions.
```
[the remark this answers is not in the document]
Comment (pauby) [reply CONTRIBUTING.md]: Agreed. Some of this was copied from the Chocolatey projects so we need to personalise for Boxstarter still. I'll add that.

[the remark this answers is not in the document]
Comment (pauby) [reply CONTRIBUTING.md]: Done!

**Pull request merged**